domain/removal/state/model/application.go

```diff
@@ -445,6 +448,42 @@ WHERE device_constraint_uuid IN (
 	return nil
 }
 
+func (st *State) deleteApplicationResources(ctx context.Context, tx *sqlair.TX, aUUID string) error {
+	appID := entityUUID{UUID: aUUID}
+
+	getApplicationResourcesStmt, err := st.Prepare(`
+SELECT resource_uuid AS &entityUUID.uuid
+FROM application_resource
+WHERE application_uuid = $entityUUID.uuid
+`, appID)
+	if err != nil {
+		return errors.Capture(err)
+	}
+
+	deleteApplicationResourceStmt, err := st.Prepare(`
+DELETE FROM application_resource
+WHERE application_uuid = $entityUUID.uuid
+`, appID)
+	if err != nil {
+		return errors.Capture(err)
+	}
```

Comment:
This statement doesn't seem to need the resource UUIDs retrieved from the stmt above, why not leave it in the `deleteSimpleApplicationReferences()`?

Reply:
Got it, the only possible working order is:
READ app resources -> DEL app resources -> DEL resources